apsuite/commissioning_scripts/calc_tune_matrix.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python-sirius
+"""."""
+
+from copy import deepcopy as _dcopy
+import numpy as np
+from pymodels import bo, si
+import pyaccel
+
+
+class Tune():
```

Comment:
I think it is important to generalize this, allowing the user to define which quadrupole families will be used for correction. For example, this class should be generic enough to allow the user to correct the tunes with dipole families, or sextupole families... We cannot predict the case of use of such a general purpose class.

In this line of thought, this class miss a property with the name of the quadrupole families which will be used and also, maybe optional initialization attributes with the desired quadrupole families.

Besides, it is important to reproduce in this class the proportional algorithm which is used in our tune correction IOC. This method groups QDs and QFs in groups and apply DeltaKs proportional to the strength of the given family. To accomplish this in a generic way it would be nice to have properties to set the QDs and QFs in two groups...

Reply:
Indeed. I believe that the changes I've made so far are converging to the idea of using generic elements as knobs for tune correction (for example, setting in the initialization the input knobs_names as a list with magnets names to be used).

The proportional algorithm implementation is on top of my to-do list.

Reply:
done!